### PR TITLE
chore: force MA0004 rule - configreawait(false) check #1451

### PR DESCRIPTION
**Pull request by frolyo**

### CODE REVIEW

File: src/dotnet/Chat.UI.Blazor/Services/ChatListUI.cs [hunk 1]
```diff
@@ -118,6 +119,7 @@ public virtual async Task<Trimmed<int>> GetUnreadChatCount(ChatListFilter filter
     }
 
     [ComputeMethod]
+    [SuppressMessage("Usage", "MA0004:Use Task.ConfigureAwait(false)")]
```
Comment (crui3er) [inline src/dotnet/Chat.UI.Blazor/Services/ChatListUI.cs]: Why do we need this SuppressMessageAttribute? Is not better to use `await ActiveChatsUI.WhenLoaded.ConfigureAwait(true)` if we need to continue after with the same synchronization context.

File: src/dotnet/Chat.UI.Blazor/Services/OnboardingUI.cs [hunk 2]
```diff
@@ -53,7 +53,7 @@ public async Task<bool> TryShow()
         // We give it 5 seconds to complete, otherwise it won't be shown
         using var cts = _lastTryShowCts = new CancellationTokenSource(TimeSpan.FromSeconds(5));
         try {
-            shouldBeShown = await ShouldBeShown(cts.Token);
+            shouldBeShown = await ShouldBeShown(cts.Token).ConfigureAwait(false);
```
Comment (crui3er) [inline src/dotnet/Chat.UI.Blazor/Services/OnboardingUI.cs]: Should not we use `ConfigureAwait(true)` here to invoke `ModalUI.Show` later in the the original context?

Comment (alexyakunin) [reply src/dotnet/Chat.UI.Blazor/Services/OnboardingUI.cs]: Yep.

File: src/dotnet/Notification.UI.Blazor/WebDeviceTokenRetriever.cs [hunk 3]
```diff
@@ -2,16 +2,13 @@
...
 
-    private IJSRuntime JS { get; }
-
-    public WebDeviceTokenRetriever(IJSRuntime js)
-        => JS = js;
+    private IJSRuntime JS { get; } = services.GetRequiredService<IJSRuntime>();
 
     public async Task<string?> GetDeviceToken(CancellationToken cancellationToken)
-        => await JS.InvokeAsync<string?>(JSGetDeviceTokenMethod, cancellationToken);
+        => await JS.InvokeAsync<string?>(JSGetDeviceTokenMethod, cancellationToken).ConfigureAwait(false);
```
Comment (alexyakunin) [inline src/dotnet/Notification.UI.Blazor/WebDeviceTokenRetriever.cs]: Use `.AsTask()` instead?

File: src/dotnet/UI.Blazor/AppBlazorCircuitContext.cs [hunk 4]
```diff
@@ -46,7 +46,7 @@ protected override void Dispose(bool disposing)
         async Task DelayedDispose()
         {
             // We want it to use the same scheduler everywhere
-            await Clocks.CpuClock.Delay(TimeSpan.FromSeconds(10));
+            await Clocks.CpuClock.Delay(TimeSpan.FromSeconds(10), StopToken);
```
Comment (alexyakunin) [inline src/dotnet/UI.Blazor/AppBlazorCircuitContext.cs]: !!! No, `StopToken` shouldn't be passed here - it's already cancelled at this point, so the rest of method won't run!

Comment (alexyakunin) [reply src/dotnet/UI.Blazor/AppBlazorCircuitContext.cs]: Please leave a comment about this as well.

File: src/dotnet/Chat.UI.Blazor/Services/ChatListUI.cs (hunk 1 above)
Comment (alexyakunin) [reply src/dotnet/Chat.UI.Blazor/Services/ChatListUI.cs]: Agree with Dima, `.ConfigureAwait(true)` is the right way to address this - it indicates this is intended.

File: src/dotnet/UI.Blazor/Services/ModalUI.cs [hunk 5]
```diff
@@ -37,7 +25,7 @@ public async Task<ModalRef> Show<TModel>(TModel model, ModalOptions options)
         where TModel : class
     {
         var componentType = GetComponentType(model);
-        return await Show(componentType, model, options);
+        return await Show(componentType, model, options).ConfigureAwait(false);
```
Comment (alexyakunin) [inline src/dotnet/UI.Blazor/Services/ModalUI.cs]: -> `.AsTask()`?

File: src/dotnet/UI.Blazor/Services/ModalUI.cs [hunk 6]
```diff
@@ -48,7 +36,7 @@ public async Task<ModalRef> Show<TModel>(TModel model, ModalOptions options)
         ModalOptions options)
         where TModel : class
     {
-        await WhenReady;
+        await WhenReady.ConfigureAwait(false);
```
Comment (alexyakunin) [inline src/dotnet/UI.Blazor/Services/ModalUI.cs]: The code below seem to require synchronization context.

File: src/dotnet/UI.Blazor/Services/ReloadUI.cs [hunk 7]
```diff
@@ -21,7 +21,7 @@ public virtual void Reload(bool clearCaches = false)
             Log.LogWarning("Reloading...");
             try {
                 if (clearCaches)
-                    await ClearCaches();
+                    await ClearCaches().ConfigureAwait(false);
```
Comment (alexyakunin) [inline src/dotnet/UI.Blazor/Services/ReloadUI.cs]: `NavigationManager` is used further, so -> `.ConfigureAwait(true)`.

File: src/dotnet/UI.Blazor/Services/TuneUI/TuneUI.cs [hunk 8]
```diff
@@ -53,7 +53,7 @@ private async ValueTask InvokeInit()
     {
         try
         {
```
Comment (alexyakunin) [inline src/dotnet/UI.Blazor/Services/TuneUI/TuneUI.cs]: Should be on the same line with `try`

File: src/dotnet/UI.Blazor/UIEvents/UIEventHub.cs [hunk 9]
```diff
@@ -1,5 +1,8 @@
+using System.Diagnostics.CodeAnalysis;
+
 namespace ActualChat.UI.Blazor;
 
+[SuppressMessage("Usage", "MA0004:Use Task.ConfigureAwait(false)")]
```
Comment (alexyakunin) [inline src/dotnet/UI.Blazor/UIEvents/UIEventHub.cs]: Seems unnecessary.

**Review by alexyakunin — CHANGES_REQUESTED** [review]

File: src/dotnet/Chat.UI.Blazor/Services/ChatListUI.cs (hunk 1 above)
Comment (frolyo) [reply src/dotnet/Chat.UI.Blazor/Services/ChatListUI.cs]: done

File: src/dotnet/Chat.UI.Blazor/Services/OnboardingUI.cs (hunk 2 above)
Comment (frolyo) [reply src/dotnet/Chat.UI.Blazor/Services/OnboardingUI.cs]: done

File: src/dotnet/Notification.UI.Blazor/WebDeviceTokenRetriever.cs (hunk 3 above)
Comment (frolyo) [reply src/dotnet/Notification.UI.Blazor/WebDeviceTokenRetriever.cs]: done

File: src/dotnet/UI.Blazor/AppBlazorCircuitContext.cs (hunk 4 above)
Comment (frolyo) [reply src/dotnet/UI.Blazor/AppBlazorCircuitContext.cs]: done

File: src/dotnet/UI.Blazor/Services/ModalUI.cs (hunk 5 above)
Comment (frolyo) [reply src/dotnet/UI.Blazor/Services/ModalUI.cs]: done

File: src/dotnet/UI.Blazor/Services/ModalUI.cs (hunk 6 above)
Comment (frolyo) [reply src/dotnet/UI.Blazor/Services/ModalUI.cs]: done

File: src/dotnet/UI.Blazor/Services/ReloadUI.cs (hunk 7 above)
Comment (frolyo) [reply src/dotnet/UI.Blazor/Services/ReloadUI.cs]: done

File: src/dotnet/UI.Blazor/Services/TuneUI/TuneUI.cs (hunk 8 above)
Comment (frolyo) [reply src/dotnet/UI.Blazor/Services/TuneUI/TuneUI.cs]: done

File: src/dotnet/UI.Blazor/UIEvents/UIEventHub.cs (hunk 9 above)
Comment (frolyo) [reply src/dotnet/UI.Blazor/UIEvents/UIEventHub.cs]: removed

**Pull request merged**